pkg/sql/pg_catalog_test.go

```diff
@@ -28,32 +28,77 @@
 package sql
 
 import (
+	"bufio"
 	"context"
 	"encoding/json"
 	"flag"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/errors/oserror"
+	"github.com/lib/pq/oid"
 )
 
 // Test data files
 const (
-	pgCatalogDump = "pg_catalog_tables.json"              // PostgreSQL pg_catalog schema
-	expectedDiffs = "pg_catalog_test_expected_diffs.json" // Contains expected difference between postgres and cockroach
-	testdata      = "testdata"                            // testdata directory
+	pgCatalogDump   = "pg_catalog_tables.json"              // PostgreSQL pg_catalog schema
+	expectedDiffs   = "pg_catalog_test_expected_diffs.json" // Contains expected difference between postgres and cockroach
+	testdata        = "testdata"                            // testdata directory
+	catalogPkg      = "catalog"
+	catconstantsPkg = "catconstants"
+	constantsGo     = "constants.go"
+	vtablePkg       = "vtable"
+	pgCatalogGo     = "pg_catalog.go"
+)
+
+// strings used on constants creations and text manipulation
+const (
+	pgCatalogPrefix          = "PgCatalog"
+	pgCatalogIDConstant      = "PgCatalogID"
+	tableIDSuffix            = "TableID"
+	tableDefsDeclaration     = `tableDefs: map[descpb.ID]virtualSchemaDef{`
+	tableDefsClosure         = `},`
+	allTableNamesDeclaration = `allTableNames: buildStringSet(`
+	allTableNamesClose       = `),`
+	virtualTablePosition     = `// typOid is the only OID generation approach that does not use oidHasher, because`
+	virtualTableTempl        = `var %s = virtualSchemaTable{
```

Comment:
Replace with virtualTableTemplate for clarity

Reply:
Done.